src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -515,16 +543,26 @@ private boolean verifySharingPermissions(String user, Path path) {
                         if (isVerified) {
                             try {
                                 // can overwrite?
-                                fs.write(path, new byte[]{read[0]});
-                            } catch (Exception ex) {
-                                LOG.log(Level.SEVERE, "User " + sharee + " could not write  shared-path " + path + "!", ex);
-                                isVerified = false;
```

Comment:
Can you explain what the logic for this is?

Reply:
working on it. The above solves the problem, but why the snapshot is out-of-date is not answered yet.

Reply:
removed - was due to simulator not taking into account existing sharing state